cpp/src/merge/merge.cu

```diff
@@ -107,36 +105,35 @@ __global__ void materialize_merged_bitmask_kernel(
 
 void materialize_bitmask(column_view const& left_col,
                          column_view const& right_col,
-                         mutable_column_view& out_col,
+                         bitmask_type* out_validity,
+                         size_type out_validity_size,
```

Comment:
I think `out_validity_size` is going to be 8x the size of the array pointed to by `out_validity`, but the naming suggests that `*(out_validity + out_validity_size)` would deference the last element of the array, when in fact it would be OOB/UB. Is there a better name we can use? `num_elements` comes to mind.